### PR TITLE
Port zigbee cli zcl ping command [KRKNWK-7706]

### PR DESCRIPTION
**Pull request by sebastiandraus**

### CODE REVIEW

File: subsys/zigbee/cli/zigbee_cli_cmd_ping.c [hunk 1]
```diff
@@ -0,0 +1,811 @@
...
+	if (ping_entry->ping_req_data.count > PING_MAX_LENGTH) {
+		if (ping_entry->ping_req_data.cb) {
+			ping_entry->ping_req_data.cb(PING_EVT_ERROR, 0,
+						     ping_entry);
+		}
+		return;
+	}
+
+	/* Make sure ZBOSS buffer API is called safely. */
+	zb_osif_disable_all_inter();
```
[the remark this answers is not in the document]
Comment (tomchy) [reply subsys/zigbee/cli/zigbee_cli_cmd_ping.c:363]: Unfortunately the thread- (not interrupt-) safe buffer API is enabled by the `ZB_INTERRUPT_SAFE_BUFS` configuration macro and it is not defined nor configurable in our `zb_vendor.h` file. I think we should add with the next ZBOSS update.

BTW: this name is a bit misleading in NCS case. The `zb_osif_disable_all_inter` API is not disabling all interrupts, but trying to get the ZBOSS mutex, so effectively in this case this implementation is identical to the effect of the `ZB_INTERRUPT_SAFE_BUFS` config.

**Review by jakub-uC — APPROVED** [review]
LGTM from shell perspective

**Comment by wbober** [comment]
Docs missing.

File: subsys/zigbee/CMakeLists.txt [hunk 2]
```diff
@@ -34,6 +34,8 @@ zephyr_library_sources_ifdef(
 	cli/zigbee_cli.c
```
Comment (rlubos) [inline subsys/zigbee/CMakeLists.txt:34]: `...` in commit 787fefda87d1446cf1774a80224cad588e14e217 is not a valid description. I suggest squashing the first two commits before merge anyway, there's no need to introduce non-building SDK code into the tree.

File: subsys/zigbee/cli/zigbee_cli_utils.c [hunk 3]
```diff
@@ -15,8 +15,11 @@
 #include <zigbee_logger_eprxzcl.h>
 #include "zigbee_cli_utils.h"
 
+extern zb_uint8_t cli_agent_ep_handler_ping(zb_bufid_t bufid);
```
Comment (rlubos) [inline subsys/zigbee/cli/zigbee_cli_utils.c:18]: It's a function declaration, `extern` is implicit here.

File: subsys/zigbee/cli/zigbee_cli_ping_types.h [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+#define PING_CUSTOM_CLUSTER           0xBEEF
+#define PING_MAX_LENGTH               79
+#define PING_ECHO_REQUEST             0x00
+#define PING_ECHO_REPLY               0x01
+#define PING_ECHO_NO_ACK_REQUEST      0x02
+#define PING_NO_ECHO_REQUEST          0x03
+#define PING_ECHO_REQUEST_BYTE        0xAB
+#define PING_ECHO_REPLY_BYTE          0xCD
+#define PING_ECHO_REQUEST_TIMEOUT_S   10
+// #define PING_TABLE_SIZE               10
```
Comment (rlubos) [inline subsys/zigbee/cli/zigbee_cli_ping_types.h:25]: Just remove it if it's not needed.

File: subsys/zigbee/cli/zigbee_cli_ping_types.h [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+/**@brief  Ping event callback definition.
+ *
+ * @param[in] evt           Type of received  ping acknowledgment
+ * @param[in] delay_ms      Time, in miliseconds, between ping request
+ *                          and the event.
+ * @param[in] entry         Pointer to context manager entry in which the ping
+ *                          request data is stored.
+ */
+typedef void (*ping_time_cb)(enum ping_time_evt evt, zb_uint32_t delay_ms,
```
Comment (rlubos) [inline subsys/zigbee/cli/zigbee_cli_ping_types.h:60]: `ping_time_cb_t`

**Review by greg-fer — CHANGES_REQUESTED** [review]
Please update documentation, I guess lib_zigbee_shell.rst.

**Comment by rlubos** [comment]
Please also add a changelog entry for this change (it's now required to update https://github.com/nrfconnect/sdk-nrf/blob/master/doc/nrf/releases/release-notes-latest.rst within a PR).

**Review by greg-fer — APPROVED** [review]

**Review by wbober — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**